### PR TITLE
ISPN-10724 Ignored caches per server causes a performance regression

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-10724

### CODE REVIEW

**Comment by gustavonalle** [comment]
run performance tests please
cs-hotrod dist writes

**Comment by diegolovison** [comment]
run performance tests please
cs-hotrod dist writes

**Comment by diegolovison** [comment]
run performance tests please
cs-hotrod dist writes

**Comment by diegolovison** [comment]
run performance tests please
cs

**Comment by gustavonalle** [comment]
Perf tests look good to me, some benchmarks show better perf than JDG 7.3.1 or slightly worse, but hard to tell if this is noise or not

**Comment by gustavonalle** [comment]
run performance tests please
cs

**Comment by gustavonalle** [comment]
run performance tests please
cs

**Comment by gustavonalle** [comment]
run performance tests please
cs

**Comment by gustavonalle** [comment]
run performance tests please
cs

**Comment by gustavonalle** [comment]
run performance tests please
cs

**Comment by gustavonalle** [comment]
I collected the results for all perf tests executed above, to figure out a supposedly perf drop on HotRod DIST writes compared to ```master``` before the 1st Igored Cache PRs:


Test | Throughput 10 threads | Mean 10 Threads | Throughput 100 Threads | Mean 100 Threads 
-- | -- | -- | -- | -- | 
Lookup | 16500 | 4709966 | 18701 | 42464694
Lookup |  19768 | 3927914 | 22975 | 34585711
Boolean | 21604 | 3597446 | 26397 | 30100640
Boolean |  17033 | 4556401 | 18954 | 41935692
Boolean | 20094 | 3841325 | 23153 | 34295585
ShortCircuit | 21765 | 3552460 | 26512 | 29915599
ShortCircuit |  20228 | 3846750 | 23368 | 33999102

Where ```Lookup``` is when reading from a ConcurrentSet (always empty), ```Boolean``` is bypassing the Set lookup if it's empty, and ```ShortCircuit``` is always returning  a hardcoded ```false``` for the ```isCacheIgnored``` method. This PR settled in the ```Boolean``` method of checking for ignored cache on remote requests.

**Comment by danberindei** [comment]
Integrated, thanks Gustavo!

We'll have to look further into PerfAck and see why the numbers are so different when the code didn't change.

File: server/core/src/main/java/org/infinispan/server/core/CacheIgnoreManager.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import org.infinispan.notifications.cachelistener.annotation.CacheEntryModified;
+import org.infinispan.notifications.cachelistener.event.CacheEntryCreatedEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryModifiedEvent;
+
+/**
+ * Manages the cache blacklisting for a given {@link EmbeddedCacheManager}
+ *
+ * @since 10.0
+ */
+public final class CacheIgnoreManager {
```
Comment (anistor) [inline server/core/src/main/java/org/infinispan/server/core/CacheIgnoreManager.java]: A beautifully mindboggling name :)))

**Pull request merged**